src/main/java/org/folio/circulation/infrastructure/storage/inventory/ItemRepository.java

```diff
@@ -377,10 +377,11 @@ private ItemRepository(Clients clients, LocationRepository locationRepository,
         records.getTotalRecords()));
   }
 
-  public CompletableFuture<Result<Collection<Item>>> findByQuery(Result<CqlQuery> queryResult) {
-    FindWithCqlQuery<Item> fetcher = RecordFetching.findWithCqlQuery(itemsClient, ITEMS_COLLECTION_PROPERTY_NAME, Item::from);
+  public CompletableFuture<Result<Collection<Item>>> findByIndexNameAndHoldingsIds(String indexName, Collection<String> ids) {
```

Comment:
Is this method specific to holdings as the name suggests or general, as the parameters (and implementation) suggest?

Reply:
Actually the method is general

Reply:
Could it be renamed to reflect that it is general and does not rely on the IDs being for holdings records?

```suggestion
  public CompletableFuture<Result<Collection<Item>>> findBy(String indexName, Collection<String> ids) {
```

Reply:
There are many names already in use. Thanks for your suggestion, it is renamed.